cni/pkg/zdsapi/zds.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// GRPC package - part of the URL. Service is added.
+// URL: /PACKAGE.SERVICE/METHOD
+package istio.workload.zds;
+
+option go_package="pkg/zdsapi";
+
+// Add a workload to the ztunnel. Ancillary data containing
+// the workload's netns file descriptor will be appended to the end of this message.
+message AddWorkload {
+  string uid = 2;
+}
+
+// Keep workload that we can't find in the fd cache. This can only be sent before SnapshotSent is sent
```

Comment:
I cannot really grok when this would happen. How can we have a pod running that we cannot find the netns?

Reply:
This being extra cautious. Our methods of resolving the netns for already running pods is heuristic, so out of extra caution, we added this option

Reply:
I don't think we've actually reproduced a scenario where it is used. We may not need it.

Reply:
we didn't reproduce it, but unless we have a method of restoring the cache that's not heuristic (which i think we don't have based on discussions in a different comment) I prefer to keep this. Maybe even add a metric for when we use it so we know if its really a problem

Reply:
> Our methods of resolving the netns for already running pods is heuristic

In the sense that for some CNI we haven't encountered yet it may not work at all, but not in the sense that the heuristic would fail and then later succeed for the same pod, right?

The latter case is what this is trying to cover.

Reply:
Yes, that is the assumption - a temporary failure with the heuristic will eventually work itself out (otherwise ztunnel restarts won't work).
This is meant to cover the case of agent restart when the ztunnel was not restarted, so the cni plugin add path does not apply here

Reply:
Yep, I don't see how the heuristic could *temporarily* fail though - it will permanently fail to find the netns, or not fail to find the netns - there's no race there post-CNI.

Reply:
In theory i agree with you, but I'm not sure if we cant prove that there is no race. The scenario i thought about is that we restart just as a pod is being started and doesn't have any processes in the proc yet.

Reply:
> The scenario i thought about is that we restart just as a pod is being started and doesn't have any processes in the proc yet.

If agent restarts as a pod is being (re)started, the plugin should block the pod until the agent responds. There shouldn't be a restart race here.

Reply:
my thinking was:
- pod starts
- cni runs and everything is good with ztunnel and it sends an ack
- now cni agent restarts
at this point, i don't know if the pod processes started and are in the `/proc`, or is it still in the process of initializing.

i know it's far reaching... i figured better safe than sorry. I'd feel a lot more comfortable to remove this if our cache re-hydration wouldn't be heuristic. I have some ideas here, but probably better for a different PR